storage/src/main/java/org/apache/kafka/storage/internals/log/TimeIndex.java

```diff
@@ -74,13 +74,13 @@ public void sanityCheck() {
         TimestampOffset entry = lastEntry();
         long lastTimestamp = entry.timestamp;
         long lastOffset = entry.offset;
+        if (entries() != 0 && lastOffset < baseOffset())
+            throw new CorruptIndexException("Corrupt time index found, time index file (" + file().getAbsolutePath() + ") has "
+                    + "non-zero size but the last offset is " + lastOffset + " which is less than the first offset " + baseOffset());
```

Comment:
This is a nice change. We should fail fast by checking local state even before calling the expensive mmap() function below.